File: aws/productivo/call_lambda.js
import { LambdaClient, InvokeCommand } from "@aws-sdk/client-lambda";

const client = new LambdaClient({
  region: "us-east-1"
});

async function llamarLambda() {
  const payload = {
  "Records": [
    {
      "cf": {
        "config": {
          "distributionId": "EXAMPLE"
        },
        "request": {
          "uri": "/test",
          "method": "GET",
          "clientIp": "2001:cdba::3257:9652",
          "headers": {
            "host": [
              {
                "key": "Host",
                "value": "d123.cf.net"
              }
            ]
          }
        }
      }
    }
  ]
};

  const command = new InvokeCommand({
    FunctionName: "Ajolo-T",
    InvocationType: "RequestResponse",
    Payload: Buffer.from(JSON.stringify(payload))
  });

  const response = await client.send(command);

  const respuestaJson = JSON.parse(new TextDecoder().decode(response.Payload));
  console.log("Respuesta Lambda:", respuestaJson);
}

llamarLambda();

File: aws/productivo/call_lambda.py
import boto3
import json

# Crear cliente Lambda
lambda_client = boto3.client("lambda", region_name="us-east-1")

# Datos que quieres enviar a la lambda
payload = {
  "version": "2.0",
  "routeKey": "GET /tasks",
  "rawPath": "/tasks",
  "requestContext": {
    "http": {
      "method": "GET",
      "path": "/tasks",
      "protocol": "HTTP/1.1",
      "sourceIp": "127.0.0.1",
      "userAgent": "curl/8.5.0"
    }
  },
  "headers": {
    "authorization": "Bearer 06b09558-0dab-41e9-959d-7f0c7d046cbf",
    "host": "localhost:3000"
  },
  "isBase64Encoded": False,
  "queryStringParameters": None,
  "body": None
}


# Invocar la función
response = lambda_client.invoke(
    FunctionName="Ajolo-T",
    InvocationType="RequestResponse",  # También puede ser "Event"
    Payload=json.dumps(payload)
)

# Leer la respuesta
respuesta_json = json.loads(response["Payload"].read())
print(respuesta_json)